Detections/MultipleDataSources/GainCodeExecutionADFSviaWMI.yaml

```diff
@@ -0,0 +1,96 @@
+id: 0bd65651-1404-438b-8f63-eecddcec87b4
+name: Gain Code Execution on ADFS Server via Remote WMI Execution
+description: |
+   'This query detects instances where an attacker has gained the ability to execute code on an ADFS Server through remote WMI Execution.
+   In order to use this query you need to be collecting Sysmon EventIDs 19, 20, and 21.
+   If you do not have Sysmon data in your workspace this query will raise an error stating:
+        Failed to resolve scalar expression named "[@Name]"
+   For more on how WMI was used in Solorigate see https://www.microsoft.com/security/blog/2021/01/20/deep-dive-into-the-solorigate-second-stage-activation-from-sunburst-to-teardrop-and-raindrop/.
+   The query contains some features from the following detections to look for potentially malicious ADFS activity. See them for more details.
+   - ADFS Key Export (Sysmon): https://github.com/Azure/Azure-Sentinel/blob/master/Detections/SecurityEvent/ADFSKeyExportSysmon.yaml
+   - ADFS DKM Master Key Export: https://github.com/Azure/Azure-Sentinel/blob/master/Detections/MultipleDataSources/ADFS-DKM-MasterKey-Export.yaml'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: SecurityEvents
+    dataTypes:
+      - SecurityEvent
+queryFrequency: 1d
+queryPeriod: 1d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - LateralMovement
+relevantTechniques:
+  - T1210
+tags:
+  - Solorigate
+query: |
+  let timeframe = 1d;
+  // Adjust for a longer timeframe for identifying ADFS Servers
+  let lookback = 6d;
+  // Identify ADFS Servers
+  let ADFS_Servers = (
+  Event
+  | where TimeGenerated > ago(timeframe+lookback)
+  | where Source == "Microsoft-Windows-Sysmon"
```

Comment:
Can we filter to a specific Event ID for Sysmon, such as EventID 1 Process Creation?  This will allow the query to be more efficient.